internal/whatsapp/client.go

```
@@ -0,0 +1,366 @@
+// DownloadMedia fetches a media object's bytes from a GetMediaURL-issued CDN URL.
+func (c *Client) DownloadMedia(ctx context.Context, acc Account, mediaURL string) ([]byte, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, mediaURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("building media request: %w", err)
+	}
+	req.Header.Set("Authorization", "Bearer "+acc.AccessToken)
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("downloading media: %w", err)
+	}
+	defer resp.Body.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `resp.Body.Close` is not checked

(errcheck)